controllers/certconfigmapgenerator/certconfigmapgenerator_controller.go

```diff
@@ -0,0 +1,140 @@
+func (r *CertConfigmapGeneratorReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	configmapGenLog.Info("Adding controller for Configmap Generation.")
+	return ctrl.NewControllerManagedBy(mgr).
+		Named("cert-configmap-generator-controller").
+		Watches(&source.Kind{Type: &corev1.ConfigMap{}}, handler.EnqueueRequestsFromMapFunc(r.watchTrustedCABundleConfigMapResource), builder.WithPredicates(ConfigMapChangedPredicate)).
+		Watches(&source.Kind{Type: &corev1.Namespace{}}, handler.EnqueueRequestsFromMapFunc(r.watchNamespaceResource), builder.WithPredicates(NamespaceCreatedPredicate)).
+		WithEventFilter(predicate.Or(predicate.GenerationChangedPredicate{}, predicate.LabelChangedPredicate{})).
+		Complete(r)
+}
+
+// Reconcile will generate new secret with random data for the annotated secret
+// based on the specified type and complexity. This will avoid possible race
+// conditions when a deployment mounts the secret before it is reconciled.
+func (r *CertConfigmapGeneratorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	// Request includes namespace that is newly created or where odh-trusted-ca-bundle configmap is updated.
+	r.Log.Info("Reconciling certConfigMapGenerator.", " CertConfigMapGenerator Request.Name", req.NamespacedName)
+	// Get namespace instance
+	userNamespace := &corev1.Namespace{}
+	err := r.Client.Get(ctx, client.ObjectKey{Name: req.Namespace}, userNamespace)
+	if err != nil {
+		return ctrl.Result{}, errors.WithMessage(err, "error getting user namespace to inject trustedCA bundle ")
+	}
+
+	// Get DSCI instance
+	dsciInstances := &dsci.DSCInitializationList{}
+	err = r.Client.List(ctx, dsciInstances)
+	if err != nil {
+		r.Log.Error(err, "Failed to retrieve DSCInitialization resource for certconfigmapgenerator ", "CertConfigmapGenerator Request.Name", req.Name)
+		return ctrl.Result{}, err
+	}
+
+	var dsciInstance *dsci.DSCInitialization
+	switch len(dsciInstances.Items) {
+	case 0:
+		return ctrl.Result{}, nil
+	case 1:
+		dsciInstance = &dsciInstances.Items[0]
+	default:
+		message := "only one instance of DSCInitialization object is allowed"
+		return ctrl.Result{}, errors.New(message)
+	}
+
+	if dsciInstance.Spec.TrustedCABundle.ManagementState != operatorv1.Managed {
+		return ctrl.Result{}, nil
+	}
+
+	// Verify if namespace did not opt out of trustedCABundle injection
+	if trustedcabundle.HasCABundleAnnotationDisabled(userNamespace) {
+		err := trustedcabundle.DeleteOdhTrustedCABundleConfigMap(ctx, r.Client, req.Namespace)
+		if err != nil {
+			r.Log.Error(err, "error deleting existing configmap from namespace", "name", trustedcabundle.CAConfigMapName, "namespace", req.Namespace)
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Verify odh-trusted-ca-bundle Configmap is created for the new namespace
+	if trustedcabundle.ShouldInjectTrustedBundle(userNamespace) {
+		err = trustedcabundle.CreateOdhTrustedCABundleConfigMap(ctx, r.Client, req.Namespace, dsciInstance)
```

Comment:
can just pass the string,  no need dsci CR, 
most of the new functions passing *dsci.DSCInitialization is only to check/set/use `dscInit.Spec.TrustedCABundle.CustomCABundle ` value 
e.g 
```suggestion
       trustCAData :=  dscInit.Spec.TrustedCABundle.CustomCABundle
		err = trustedcabundle.CreateOdhTrustedCABundleConfigMap(ctx, r.Client, req.Namespace, trustCAData)
```